src/dotnet/commands/dotnet-add/dotnet-add-p2p/Program.cs

```diff
@@ -44,17 +47,51 @@ public static int Run(string[] args)
                     throw new GracefulException(CommonLocalizableStrings.RequiredArgumentNotPassed, $"<{LocalizableStrings.ProjectException}>");
                 }
 
-                var msbuildProj = MsbuildProject.FromFileOrDirectory(projectArgument.Value);
+                ProjectCollection collection = new ProjectCollection();
+                var msbuildProj = MsbuildProject.FromFileOrDirectory(collection, projectArgument.Value);
 
                 if (app.RemainingArguments.Count == 0)
                 {
                     throw new GracefulException(LocalizableStrings.SpecifyAtLeastOneReferenceToAdd);
                 }
 
+                string frameworkString = frameworkOption.Value();
                 List<string> references = app.RemainingArguments;
                 if (!forceOption.HasValue())
                 {
                     MsbuildProject.EnsureAllReferencesExist(references);
+                    IEnumerable<MsbuildProject> refs = references.Select((r) => MsbuildProject.FromFile(collection, r));
+
+                    if (frameworkString == null)
```

Comment:
Does all this check really belong in the Program.cs? Feels to me like it should be in its own abstraction.